src/main/kotlin/Board.kt

```diff
@@ -0,0 +1,108 @@
+class Board(initialPositions: Map<Char, List<String>> = mapOf()) {
+    private var turn: Int = 0
+    private val boardSpots = arrayOf(charArrayOf(' ', ' ', ' '), charArrayOf(' ', ' ', ' '), charArrayOf(' ', ' ', ' '))
+
+    init {
+        initialPositions.forEach { (xOr0, placements) ->
+            placements.forEach { move ->
+                val (row, col) = parseMove(move)
+                boardSpots[row][col] = xOr0
+            }
+        }
+    }
+
+    fun addMove(move: String) {
+        val (row, col) = parseMove(move)
+
+        if (isZerosTurn(turn)) {
+            boardSpots[row][col] = 'O'
+        } else {
+            boardSpots[row][col] = 'X'
+        }
+        turn++
+    }
+
+    private fun isZerosTurn(turn: Int) = turn % 2 == 0
+
+    fun isOccupied(move: String): Boolean {
```

Comment:
I was able to first extract this method, and then change it's implementation without changing it's public implementation. (It no longer needs to track player moves).